executor/index_merge_reader.go

```diff
@@ -591,9 +626,12 @@ func (w *partialIndexWorker) fetchHandles(ctx context.Context, result distsql.Se
 	}
 }
 
-func (w *partialIndexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (
+func (w *partialIndexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult, handleCols plannercore.HandleCols) (
 	handles []kv.Handle, retChk *chunk.Chunk, err error) {
-	handleOffset := chk.NumCols() - 1
+	var handleOffset []int
+	for i := 0; i < handleCols.NumCols(); i++ {
+		handleOffset = append(handleOffset, chk.NumCols()-handleCols.NumCols()+i)
```

Comment:
Since the index row's handle columns always at the last, we can move this logic into `HandleCols`, and use a method `BuildHandleByIndexRow`.